homeassistant/components/zeroconf/__init__.py

```diff
@@ -222,7 +224,9 @@ async def _async_register_hass_zc_service(
     # Set old base URL based on external or internal
     params["base_url"] = params["external_url"] or params["internal_url"]
 
-    host_ip = util.get_local_ip()
+    host_ip = hass.loop.run_until_complete(
+        async_get_source_ip(hass, target_ip=PUBLIC_TARGET_IP)
```

Comment:
We probably want the MDNS_TARGET_IP since thats going to be the interface we announce it on

Reply:
Isn't multicast potentially routable on all interfaces ? In such situation, `async_get_source_ip `will return the value of the first interface found, with no strict logic.

Reply:
You'll get the route they setup for multicast or if none the default route which is likely what we want in this case since a public ip address would be less likely to go to local network.

Reply:
I also think that is a bad practice connect to a multicast address.
Anyway, I change the code as requested.

Simone

Reply:
Done